### PR TITLE
Do not mention update_cell_dof_indices_cache in documentation

### PR DESCRIPTION
**Pull request by kronbichler**
Follow-up to #14068 - this should remove all mentioning of the dof indices cache in our files. While there, I also removed a `friend` declaration that is no longer necessary after #14068.

### CODE REVIEW

File: include/deal.II/dofs/dof_levels.h [hunk 1]
```diff
@@ -38,14 +38,8 @@ namespace internal
...
      *
-     * We store are cached values for the DoF indices on each cell
-     * in#cell_dof_indices_cache, since this is a frequently requested
-     * operation. The values are set by
-     * DoFCellAccessor::update_cell_dof_indices_cache and are used by
-     * DoFCellAccessor::get_dof_indices.
-     *
      * Note that vertices are separate from, and in fact have nothing to do
-     * with cells. The indices of degrees of freedom located on vertices
+     * with, cells. The indices of degrees of freedom located on vertices
```
Comment (peterrum) [inline include/deal.II/dofs/dof_levels.h:42]: Is this new comma here correct? I would have put "in fact" between commas.

**Review by peterrum — APPROVED** [review]

File: include/deal.II/dofs/dof_levels.h (hunk 1 above)
Comment (kronbichler) [reply include/deal.II/dofs/dof_levels.h:42]: I guess it depends on how you read the sentence. I can understand your reasoning. Mine was that if you put the comma before "and", one needs to make the subordinate clause all the way to its end, i.e., after "with". Either way, this sentence feels a bit too wrinkled (German?), so let me rephrase this a bit more.

Comment (kronbichler) [reply include/deal.II/dofs/dof_levels.h]: How about the new version?

**Review by drwells — APPROVED** [review]

**Review by masterleinad — APPROVED** [review]
